File: ANGEN/SampleFramework.Controls/NavigationBar/Samples/Snap UI Category/Fake IconEditor/VB/FakeIconEditor/PaletteForm.vb
Friend NotInheritable Class PaletteForm
    Public Event GradientBeginChanged As EventHandler(Of ColorEventArgs)
    Public Event GradientEndChanged As EventHandler(Of ColorEventArgs)

    Private Sub _colorSelector_MouseClick(ByVal sender As Object, ByVal e As MouseEventArgs) Handles _colorSelector.MouseClick
        If (e.Button = MouseButtons.Left) Then
            RaiseEvent GradientBeginChanged(Me, New ColorEventArgs(_colorSelector.SelectedColor))
        ElseIf (e.Button = MouseButtons.Right) Then
            RaiseEvent GradientEndChanged(Me, New ColorEventArgs(_colorSelector.SelectedColor))
        End If
    End Sub

    Private Sub _colorSelector_SelectedColorChanged(ByVal sender As Object, ByVal e As EventArgs) Handles _colorSelector.SelectedColorChanged
        _colorHistory.SelectedColor = _colorSelector.SelectedColor
    End Sub
End Class

File: ANGEN/SampleFramework.Controls/NavigationBar/Samples/Snap UI Category/Fake IconEditor/CS/FakeIconEditor/PaletteForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace FakeIconEditor
{
	internal sealed partial class PaletteForm : Form
	{
		public event EventHandler<ColorEventArgs> GradientBeginChanged;

		private void OnGradientBeginChanged(ColorEventArgs e)
		{
			if (this.GradientBeginChanged != null)
			{
				this.GradientBeginChanged(this, e);
			}
		}

		public event EventHandler<ColorEventArgs> GradientEndChanged;

		private void OnGradientEndChanged(ColorEventArgs e)
		{
			if (this.GradientEndChanged != null)
			{
				this.GradientEndChanged(this, e);
			}
		}

		private void _colorSelector_MouseClick(object sender, MouseEventArgs e)
		{
			if (e.Button == MouseButtons.Left)
			{
				this.OnGradientBeginChanged(new ColorEventArgs(_colorSelector.SelectedColor));
			}
			else if (e.Button == MouseButtons.Right)
			{
				this.OnGradientEndChanged(new ColorEventArgs(_colorSelector.SelectedColor));
			}
		}

		private void _colorSelector_SelectedColorChanged(object sender, EventArgs e)
		{
			_colorHistory.SelectedColor = _colorSelector.SelectedColor;
		}

		public PaletteForm()
		{
			this.InitializeComponent();
		}
	}
}
